config/task-monitor-result.yaml

```diff
@@ -0,0 +1,71 @@
+# Monitor the pipelinerun status and update the pull request
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata:
+  name: monitor-result-task
+spec:
+  inputs:
+    resources:
+      - name: pull-request
+        type: pullRequest
+    params:
+      - name: pipelinerun
+        description: The name of the pipelinerun to be monitored
+        default: default pipelinerun
+      - name: commentsuccess
+        description: The text of the sccess comment
+        default: default success at task
+      - name: commentfailure
+        description: The text of the failure comment
+        default: default failure at task
+  outputs:
+    resources:
+      - name: pull-request
+        type: pullRequest
+  steps:
+  - name: check
+    image: ibmcom/microclimate-utils:1905
+    env:
+      - name: PIPELINERUN
+        value: ${inputs.params.pipelinerun}
+      - name: COMMENT_SUCCESS
+        value: ${inputs.params.commentsuccess}
+      - name: COMMENT_FAILURE
+        value: ${inputs.params.commentfailure}
+      - name: NAMESPACE
+        valueFrom:
+          fieldRef:
+            fieldPath: metadata.namespace
+    command: ["/bin/bash"]
+    args:
+    - -ce
+    - |
+      set -e
+      apk add python
+      cat <<EOF | python
+      import subprocess
+      import json
+      import os
+      import time
+      comment = "$COMMENT_FAILURE"
+      pipelinerun = "$PIPELINERUN"
+      i = range(100)
+      for x in i:
+          time.sleep( 10 )
+          p = subprocess.Popen('kubectl get pipelinerun ' + pipelinerun + ' -n $NAMESPACE -o=jsonpath="[{.status.conditions[0].status},{.status.conditions[0].type}]"', shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```

Comment:
this is pretty gross... really not a decent easy way?! I don't like this entire huge script here really

Reply:
I try to make it a little bit neater by using the python kubernetes client library instead of the kubectl executable.

Reply:
I cleaned a little bit.